airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/GlueOperations.java

```diff
@@ -0,0 +1,156 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+//TODO (itaseski) implement wrapper for retry logic on transient errors
+public class GlueOperations implements MetastoreOperations {
+
+    private final ObjectMapper objectMapper;
+
+    private final AWSGlue awsGlueClient;
+
+    public GlueOperations(AWSGlue awsGlueClient) {
+        Preconditions.checkArgument(awsGlueClient != null);
+        this.awsGlueClient = awsGlueClient;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    //TODO (itaseski) can location change after table is created?
+    @Override
+    public void upsertTable(String databaseName, String tableName, String location, JsonNode jsonSchema) {
+        try {
+            GetTableRequest getTableRequest = new GetTableRequest()
+                .withDatabaseName(databaseName)
+                .withName(tableName);
+
+            // Will throw EntityNotFoundException if table doesn't exist
+            awsGlueClient.getTable(getTableRequest);
+
+            UpdateTableRequest updateTableRequest = new UpdateTableRequest()
+                .withDatabaseName(databaseName)
+                //TODO (itaseski) do we need to send all table inputs or the ones that are going to be changed?
+                .withTableInput(
+                    new TableInput()
+                        .withName(tableName)
+                        .withStorageDescriptor(
+                            new StorageDescriptor()
+                                .withLocation(location)
+                                .withColumns(transformSchema(jsonSchema))
+                        )
+                );
+
+            awsGlueClient.updateTable(updateTableRequest);
+        } catch (EntityNotFoundException enfe) {
+            CreateTableRequest createTableRequest = new CreateTableRequest()
+                .withDatabaseName(databaseName)
+                .withTableInput(
+                    new TableInput()
+                        .withName(tableName)
+                        .withTableType("GOVERNED")
+                        .withStorageDescriptor(
+                            new StorageDescriptor()
+                                .withLocation(location)
+                                .withColumns(transformSchema(jsonSchema))
+                                .withInputFormat("org.apache.hadoop.mapred.TextInputFormat")
+                                .withOutputFormat("org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat")
+                                .withSerdeInfo(
+                                    new SerDeInfo()
+                                        .withSerializationLibrary("org.openx.data.jsonserde.JsonSerDe")
```

Comment:
This will need to be mapped to the file type (e.g. JSON vs. Avro vs. Parquet) so that the query engine can use the proper library when processing the data.

Reply:
Yeah, I'm aware of this. but I though that we agreed that in v1 we will support only JSON and I'm performing that check in order to decide whether to generate the schema in Glue or not.